File: libraries/rng/lcg/lcg.f90

function lcg(seed)
  implicit none
  integer,parameter :: ik=selected_int_kind(15),&
                     & rk=selected_real_kind(10,40)
  real(rk) :: lcg
  integer(ik),intent(inout) :: seed
  integer(ik),save :: a=69069,c=1,m=4294967296
  real(rk),save :: rm=4294967296.0

  seed= mod(seed*a+c,m)
  lcg=seed/rm

end function lcg
  

File: libraries/rng/lcg/lcg.c

double lcg(long long int *seed)
{
  static long long int a=69069,c=1,m=4294967296; /* 2^32 */
  static double rm=4294967296.0;

  *seed=(*seed * a+c)%m;
  return (double)*seed/rm;
}
